pinot-spi/src/main/java/org/apache/pinot/spi/trace/Tracing.java

```diff
@@ -301,6 +327,7 @@ public static void clear() {
     public static void initializeThreadAccountant(PinotConfiguration config, String instanceId,
         InstanceType instanceType) {
       String factoryName = config.getProperty(CommonConstants.Accounting.CONFIG_OF_FACTORY_NAME);
+      _workloadBudgetManager = new WorkloadBudgetManager(config);
```

Comment:
Why is `_workLoadBudgetManager` instantiated here. IIUC it is used only in `WorkloadAggregator`. Can it be initialized in the the `WorkloadAggregator` ?

Reply:
Also `Tracing.java` doesn't need to know about it as well. It is specific to `ResourceUsageAccountantManager`

Reply:
Initialized it here because it needs to be accessed from QuerySchedulers and other places. Of course it can be moved to WorkloadAggregator and indirectly accessed from there. But given that we are going to eventually replace `PerQuery..Acountant`, it makes sense to keep it here?

Reply:
We discussed this offline.  Summarizing here:

- We'll be exercising caution and ensure that `ResourceUsage..Acountant` has feature parity and stability before deprecating `PerQuery...Accountant`
- Moved WorkloadBudgetManager to core. 
- WorkloadBudgetManager is not initialized. It fast exits if the config for workload budget collection is not enabled.